libcloud/storage/drivers/google_storage.py

```diff
@@ -17,116 +17,162 @@
 import copy
 import hmac
 
-from email.utils import formatdate
+import email.utils
 from hashlib import sha1
 
-from libcloud.utils.py3 import b
-
-from libcloud.common.base import ConnectionUserAndKey
-
+from libcloud.common.google import GoogleAuthType
+from libcloud.common.google import GoogleBaseConnection
 from libcloud.storage.drivers.s3 import BaseS3StorageDriver, S3Response
 from libcloud.storage.drivers.s3 import S3RawResponse
+from libcloud.utils.py3 import b
 
+# Docs are a lie. Actual namespace returned is different that the one listed
+# in the docs.
 SIGNATURE_IDENTIFIER = 'GOOG1'
-
-# Docs are a lie. Actual namespace returned is different that the one listed in
-# the docs.
-AUTH_HOST = 'commondatastorage.googleapis.com'
 API_VERSION = '2006-03-01'
 NAMESPACE = 'http://doc.s3.amazonaws.com/%s' % (API_VERSION)
 
 
-class GoogleStorageConnection(ConnectionUserAndKey):
+def _get_aws_auth_param(method, headers, params, expires, secret_key,
```

Comment:
Done. Refactored this to use the S3 HMAC Auth method. Also, there were no tests for S3's method, so I added one.